### PR TITLE
chore: check account status when generating access token and GetMe api

### PR DESCRIPTION
**Pull request by cre8ivejp**
To improve access security, we must also check the user account status when generating the access token and when the `GetMe` API is called. Currently, we only check if the account exists.

----

This pull request introduces changes to enhance the account status check and error handling in the `AccountService` and `AuthService`. The most important changes include the addition of `checkAccountStatus` method in both services, changes to `GetMe` method in `AccountService`, and updates to the corresponding test cases.

Account status check:

* [`pkg/account/api/admin_account.go`](diffhunk://#diff-1097fbb3548177580d8c5e170cc44b5cb9db843fc88c1e618802ad74122cfc95R66-R72): Added a new method `checkAccountStatus` to validate if the user account is enabled. This method is now called within the `GetMe` method to check the account status before listing projects by organization ID. [[1]](diffhunk://#diff-1097fbb3548177580d8c5e170cc44b5cb9db843fc88c1e618802ad74122cfc95R66-R72) [[2]](diffhunk://#diff-1097fbb3548177580d8c5e170cc44b5cb9db843fc88c1e618802ad74122cfc95R169-R213)

* [`pkg/auth/api/api.go`](diffhunk://#diff-aeff18bc6a7910167ad8249e1ba57c2208afc673996ab88b5718cf60baae408fR321-R327): Similarly, added a new method `checkAccountStatus` to validate if the user has at least one account enabled in any organization. This method is now called within the `generateToken` method to check the account status before generating a token. [[1]](diffhunk://#diff-aeff18bc6a7910167ad8249e1ba57c2208afc673996ab88b5718cf60baae408fR321-R327) [[2]](diffhunk://#diff-aeff18bc6a7910167ad8249e1ba57c2208afc673996ab88b5718cf60baae408fR403-R448)

Test case updates:

* [`pkg/account/api/admin_account_test.go`](diffhunk://#diff-e56903718b2dfe1cdba4377e98b8bd79e3f96e87891f34e1f4c83886509270ceR31): Updated the test cases for `GetMe` method to include scenarios where the account is not found and the account is disabled. Also, added expectations for `GetAccountV2` method in the account storage mock to return specific account statuses for these scenarios. [[1]](diffhunk://#diff-e56903718b2dfe1cdba4377e98b8bd79e3f96e87891f34e1f4c83886509270ceR31) [[2]](diffhunk://#diff-e56903718b2dfe1cdba4377e98b8bd79e3f96e87891f34e1f4c83886509270ceR85-R130) [[3]](diffhunk://#diff-e56903718b2dfe1cdba4377e98b8bd79e3f96e87891f34e1f4c83886509270ceL104-R154)

### CODE REVIEW

File: pkg/auth/api/api.go [hunk 1]
```diff
@@ -393,6 +400,52 @@ func (s *authService) checkEmail(
...
+			dt, err := auth.StatusInternal.WithDetails(&errdetails.LocalizedMessage{
+				Locale:  localizer.GetLocale(),
+				Message: localizer.MustLocalize(locale.InternalServerError),
+			})
+			if err != nil {
+				return auth.StatusInternal.Err()
+			}
+			return dt.Err()
+		}
+		if !resp.Account.Disabled {
```
Comment (cre8ivejp) [inline pkg/auth/api/api.go:433]: If the user has at least one account enabled, we allow the access token generation.

File: pkg/auth/api/api.go [hunk 2]
```diff
@@ -393,6 +400,52 @@ func (s *authService) checkEmail(
...
+			// The user must have at least one account enabled
+			return nil
```
Comment (cre8ivejp) [inline pkg/auth/api/api.go:435]: If the user has at least one account enabled, we allow the access token generation.

**Review by kentakozuka — APPROVED** [review]
LGTM 👍

**Review by Ubisoft-potato — COMMENTED** [review]
LGTM!

**Pull request merged**